src/client/api/job.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include <stdlib.h>
+#include <sys/utsname.h>
+#include <daos/common.h>
+#include <daos/job.h>
+
+
+/*
+ * The answer of what the max length of envvar name is very tricky. Arguments
+ * and environment variable share the same memory space so to make things easy
+ * we enforce an arbitrary length of 80 which some other shells enforce
+ */
+#define MAX_ENV_NAME 80
+
+#define MAX_JOBID_LEN 1024
+
+char *dc_jobid_env;
+char *dc_jobid;
+
+static int craft_default_jobid(char **jobid)
+{
+	struct utsname name = {0};
+	pid_t pid;
+	int ret;
+
+	ret = uname(&name);
+	if (ret)
+		return -DER_INVAL;
+
+	pid = getpid();
+
+	D_ASPRINTF(*jobid, "%s-%d", name.nodename, pid);
+	if (*jobid == NULL)
+		return -DER_NOMEM;
+	return 0;
+}
+
+int dc_job_init(void)
+{
+	char *jobid;
+	char *jobid_env = getenv(JOBID_ENV);
+	int err = 0;
+
+	if (jobid_env == NULL) {
+		D_STRNDUP(jobid_env, DEFAULT_JOBID_ENV,
+			  sizeof(DEFAULT_JOBID_ENV));
+	} else {
+		char *tmp_env = jobid_env;
+		D_STRNDUP(jobid_env, tmp_env, strnlen(tmp_env, MAX_ENV_NAME));
```

Comment:
(style)  Missing a blank line after declarations

Reply:
Why strnlen(tmp_env, MAX_ENV_NAME) and not just MAX_ENV_NAME here?

Reply:
That's a good point I'm not sure what I was thinking when I wrote that.